### ISSUE TITLE
Fastly backend TLS - Not enough options & ssl_check_cert no longer works

### ISSUE THREAD

**Issue by jsonmaur** [hashicorp/terraform#9003]
### Terraform Version

v0.7.4
### Affected Resource(s)
- fastly.service_v1
### Terraform Configuration Files

``` hcl
resource "fastly_service_v1" "main" {
  //...
  backend {
    address = "..."
    name = "..."
    port = 443
    ssl_check_cert = false
  }
  //...
}
```

If I have a backend on port 443 but I don't want to verify the certificate, I would think using the above would do the trick. But this results in `Error 503 hostname doesn't match against certificate` from Fastly. Looks like they changed how to do this in their service. We need to be able to set the following options correctly for `backend`, because `ssl_check_cert` doesn't work:
- Connect to backend using TLS
- Verify certificate
- Certificate Hostname
- TLS CA certificate

**Comment by zimbatm** [comment]
Can you try https://github.com/hashicorp/terraform/pull/9629 to see if it fixes your issue?

**Comment by stack72** [comment]
Closed via #9629
